File: OneToOneCallSample/app/src/main/java/com/stringee/apptoappcallsample/stringee/common/StringeeCallConfig.java
package com.stringee.apptoappcallsample.stringee.common;

/** Immutable outgoing-call options consumed by {@code StringeeCallManager}. */
public final class StringeeCallConfig {
    private final String to;
    private final CallEngine callEngine;
    private final boolean videoCall;
    private String customData;

    /** Creates a configuration for a recipient, call engine, and voice/video mode. */
    public StringeeCallConfig(String to, CallEngine callEngine, boolean videoCall) {
        this.to = to == null ? "" : to.trim();
        this.callEngine = callEngine == null ? CallEngine.STRINGEE_CALL : callEngine;
        this.videoCall = videoCall;
    }

    public String getTo() {
        return to;
    }

    public CallEngine getCallEngine() {
        return callEngine;
    }

    public boolean isVideoCall() {
        return videoCall;
    }

    public String getCustomData() {
        return customData;
    }

    /** Sets optional custom data forwarded with the call and returns this configuration. */
    public StringeeCallConfig setCustomData(String customData) {
        this.customData = customData;
        return this;
    }
}

File: Kotlin-OneToOneCallSample/app/src/main/java/com/stringee/kotlin_onetoonecallsample/stringee/common/StringeeCallConfig.kt
package com.stringee.kotlin_onetoonecallsample.stringee.common

import com.stringee.kotlin_onetoonecallsample.stringee.StringeeCallManager

/**
 * Immutable outgoing-call options consumed by [StringeeCallManager].
 *
 * @param to recipient Stringee user ID
 * @param callEngine SDK call engine to use
 * @param isVideoCall `true` for video, `false` for voice
 */
class StringeeCallConfig(to: String?, callEngine: CallEngine?, val isVideoCall: Boolean) {
    val to: String
    val callEngine: CallEngine
    var customData: String? = null
        private set

    init {
        this.to = if (to == null) "" else to.trim { it <= ' ' }
        this.callEngine = if (callEngine == null) CallEngine.STRINGEE_CALL else callEngine
    }

    /** Sets optional custom data forwarded with the call and returns this configuration. */
    fun setCustomData(customData: String?): StringeeCallConfig {
        this.customData = customData
        return this
    }
}
